src/auslib/util/rulematching.py

```diff
@@ -127,9 +127,13 @@ def matchBuildID(ruleBuildID, queryBuildID):
 def matchMemory(ruleMemory, queryMemory):
     """Decides whether a memory value from the rules matches an incoming one.
        If the ruleMemory is null, we match any queryMemory. If it's not
-       null, we must either match exactly, or match with a camparison
+       null, we must either match exactly, or match with a comparison
        operator."""
-    if ruleMemory is None:
+    if ruleMemory is None or queryMemory is None:
+        return True
+    try:
+        queryMemory = int(queryMemory)
+    except (TypeError, ValueError):
```

Comment:
At some point I think it would be good to get rid of these types of assumptions at this layer (that absent or broken data should be treated as "always true")...but this is the case for all sorts of fields right now, so this is fine.

Reply:
Yeah... I also saw that in some places we use `int()` without catching exceptions. They may need some love or refactoring. Or both :)